pkg/operations/start.go

```diff
@@ -80,6 +81,17 @@ func StartVM(vm *api.VM, debug bool) error {
 		PortBindings: vm.Spec.Network.Ports, // Add the port mappings to Docker
 	}
 
+	// Only set envVars if at least one was configured
+	var envVars []string
+	parts := strings.Split(vm.GetAnnotation((constants.IGNITE_SANDBOX_ENV_VAR)), ",")
+	for _, part := range parts {
+		if part == "" {
+			continue
+		}
+		envVars = append(envVars, part)
+	}
+	config.EnvVars = envVars
```

Comment:
Could the API be a map instead of a CSV?
I'm slightly inspired by [docker-compose's `environment`](https://docs.docker.com/compose/compose-file/compose-file-v3/#environment).

ex:
```yaml
metadata:
  annotations:
    ignite.weave.works/sanbox-env/FIRECRACKER_GO_SDK_REQUEST_TIMEOUT_MILLISECONDS: '1000'
    ignite.weave.works/sanbox-env/FIRECRACKER_GO_SDK_INIT_TIMEOUT_SECONDS: '1000'  # did you mean '1' here?
```

Code diff would iterate through the annotation map and prefix match:
(untested)
```diff
diff --git pkg/constants/vm.go pkg/constants/vm.go
index fd45b7f9..a09021fa 100644
--- pkg/constants/vm.go
+++ pkg/constants/vm.go
@@ -42,6 +42,6 @@ const (
 	// IGNITE_EXTRA_INTFS is the annotation key to store a list of extra interfaces
 	IGNITE_EXTRA_INTFS = "ignite.weave.works/extra-intfs"
 
-	// IGNITE_SANDBOX_ENV_VAR is the annotation key to store a list of env variables
-	IGNITE_SANDBOX_ENV_VAR = "ignite.weave.works/sanbox-env"
+	// IGNITE_SANDBOX_ENV_PREFIX is the annotation prefix to store a map of env variables
+	IGNITE_SANDBOX_ENV_PREFIX = "ignite.weave.works/sanbox-env/"
 )
diff --git pkg/operations/start.go pkg/operations/start.go
index 08033c74..417d40f8 100644
--- pkg/operations/start.go
+++ pkg/operations/start.go
@@ -83,12 +83,11 @@ func StartVM(vm *api.VM, debug, wait bool) error {
 
 	// Only set envVars if at least one was configured
 	var envVars []string
-	parts := strings.Split(vm.GetAnnotation((constants.IGNITE_SANDBOX_ENV_VAR)), ",")
-	for _, part := range parts {
-		if part == "" {
-			continue
+	for anno, v := range vm.GetObjectMeta().Annotations {
+		if strings.HasPrefix(anno, constants.IGNITE_SANDBOX_ENV_PREFIX) {
+			k := strings.TrimPrefix(anno, constants.IGNITE_SANDBOX_ENV_PREFIX)
+			envVars = append(envVars, fmt.Sprintf("%s=%s", k, v))
 		}
-		envVars = append(envVars, part)
 	}
 	config.EnvVars = envVars
```

Reply:
yep, got it